sql/src/main/codegen/config.fmpp

```diff
@@ -65,10 +70,16 @@ data: {
       "CLUSTERED"
       "OVERWRITE"
       "PARTITIONED"
+      "EXTERN"
+      "S3"
+      "CSV"
```

Comment:
Why make it as a keyword? If we can, let's keep it as non-reserved only.

Also, it would be cool to have tests that test the parsing if any of these keywords are in the table identifier list, something like `SELECT csv, s3, col3, col4 FROM extern` (and a few permutations of these). Queries like these will most likely break if EXTERN is a keyword (because then the parser will prevent it from being an identifier). Also I am curious what happens if these queries are run now - IMO they should pass because the current EXTERN(..) function is like a sql function and not a syntax change. 

Some more examples can be : OVERWRITE INTO EXTERN(....) SELECT col1 FROM extern WHERE....
(Here first extern refer to the new syntax, while the second one refer to the table)

If the parsing is correct, in these edge cases, we can probably make do with keeping it as non-reserved only. Else this change needs to be called out.

Reply:
These are non-reserved keywords, the way I understand it, the words need to be added to keywords, and then if it should be non reserved, also added to nonReservedKeywords (the way it's done for OVERWRITE).

Reply:
Adding a query which checks this is a good idea though. Adding a test which should fail if csv or extern are reserved keywords.